stan/math/opencl/kernel_generator/get_kernel_source_for_evaluating_into.hpp

```diff
@@ -0,0 +1,41 @@
+#ifndef STAN_MATH_OPENCL_KERNEL_GENERATOR_GET_KERNEL_SOURCE_FOR_EVALUATING_INTO_HPP  // NOLINT
+#define STAN_MATH_OPENCL_KERNEL_GENERATOR_GET_KERNEL_SOURCE_FOR_EVALUATING_INTO_HPP  // NOLINT
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/kernel_generator/operation.hpp>
+#include <stan/math/opencl/kernel_generator/as_operation.hpp>
+#include <cl.hpp>
+#include <string>
+#include <set>
+
+namespace stan {
+namespace math {
+
+template <typename Derived, typename ReturnScalar, typename... Args>
+template <typename T_lhs>
+std::string operation<Derived, ReturnScalar, Args...>::
+    get_kernel_source_for_evaluating_into(const T_lhs& lhs) const {
+  static_assert(is_valid_expression<T_lhs>::value,
+                "operation::get_kernel_source_for_evaluating_into: left hand "
+                "side is not a valid expression!");
+  auto lhs_expression = as_operation(lhs);
+  std::set<const void*> generated;
```

Comment:
Could generated be phrased as tuple of Args?

Reply:
I don't see how we could do that.

Reply:
^Pieces like this are why I want a design doc. My q here is because is because I really don't understand some key pieces here